### ISSUE TITLE
Gloo edge does not work with istio 1.13.x

### ISSUE THREAD

**Issue by kevin-shelaga** [solo-io/gloo#5989]
### Gloo Edge Version

1.10.x (latest stable)

### Kubernetes Version

1.22.x

### Describe the bug

`glooctl istio inject` and istioSDS helm values wont work with istio 1.13.x

```yml
 global:
   istioSDS:
     enabled: true
     customSidecars:
         - name: istio-proxy
           image: docker.io/istio/proxyv2:1.13.1
```

### Steps to reproduce the bug

1. install istio
2. glooctl install gateway
3. glooctl istio inject
or
install gloo with the provided yaml 

### Expected Behavior

istio mtls should work with gloo

**Comment by murphye** [comment]
From Kevin:

> Hey, so right off the bat glooctl istio inject doesn't support 1.13.x.
> 
> With a custom sidecar, it sort of seems like the mount paths for the certs changed. But there is no way to override that currently.  (edited) 
> 
> So I was dead in the water

**Comment by murphye** [comment]
Brian will check the updated cert paths from sidecar injection in Istio 1.13

**Comment by bgottfried91** [comment]
Tested `glooctl istio inject` this morning with Istio version 1.13.2 and Gloo Edge version 1.10.16. Initially ran into the issue Kevin described, where it looked like the `sds` sidecar container on the GatewayProxy pod couldn't find the certs at `/etc/istio-certs/key.pem`, as seen in the log line below:
```
{"level":"fatal","ts":"2022-03-18T16:29:40.500Z","logger":"sds_server","caller":"cmd/main.go:84","msg":"could not find file '/etc/istio-certs/key.pem'","version":"1.10.16","stacktrace":"main.main\n\t/workspace/gloo/projects/sds/cmd/main.go:84\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225"}
```

However, when I went to check the logs for my `istiod` pod, I saw this repeated error:
```
2022-03-18T16:30:01.392373Z     warn    serverca        Authentication failed for 10.244.4.13:55306: Authenticator ClientCertAuthenticator at index 0 got error: no verified chain is found. Authenticator KubeJWTAuthenticator at index 1 got error: could not get cluster Kubernetes's kube client.
```

Looking at my `istiod` deployment, I saw this environment variable:
```
        - name: CLUSTER_ID
          value: edge-testing
```
but in the sidecars that were injected into the GatewayProxy, I saw this environment variable:
```
        - name: ISTIO_META_CLUSTER_ID
          value: Kubernetes
```

Changing the `ISTIO_META_CLUSTER_ID` env to match the value in the `CLUSTER_ID` env allowed the sidecars for the GatewayProxy to start up successfully and retrieve the certs. I was then able to enable strict mTLS on the cluster, add the necessary config to my upstream and test a route to the upstream as described [here](https://docs.solo.io/gloo-edge/latest/guides/integrations/service_mesh/gloo_istio_mtls/) and everything worked as expected.

I think the `glooctl istio inject` logic needs to be looked at to determine why it's setting `ISTIO_META_CLUSTER_ID` to Kubernetes rather than the value in the `CLUSTER_ID` env for istiod, as that should hopefully resolve this issue.
